src/client/failure/policies.ts

```diff
@@ -325,7 +334,12 @@ export class CircuitBreakerPolicy extends FailurePolicy {
     if (this.state === 'half-open') {
       this.open(info.time);
 
-      return { kind: 'abort', reason: 'circuit breaker is open' };
+      return {
+        kind: 'abort',
+        reason: `Circuit breaker is open. Last error from provider: ${this.formatFailureReason(
+          info
+        )}`,
```

Comment:
So do we need both `formatFailureReason` and passing the original error in `FailurePolicyError`?

Reply:
I think we do, maybe we could do reason formatting in FailurePolicyError.
In some (pre-fetch and open circuit-breaker) cases we don't have original error so reason is all we have 🤷

Reply:
`pre-fetch` doesn't have an original error. If you are doing abort in pre-fetch it's because you are doing failover.

Open circuit-breaker either has an error that opened it right there, or it was opened last time and the shadowed by a failover. In the first case it's trivial, in the second that error wouldn't be available anyway.

Reply:
Yes, and that's why originalError is optional in FailurePolicyError and we need reason 🤷

Reply:
Yes but do we need to include the error in the reason when the error (if any) will be part of the `FailurePolicyError`?